app/services/fact_importer.rb

```diff
@@ -164,7 +164,9 @@ def update_facts
     time = Time.now.utc
     updated = 0
     db_facts_names = []
-    db_facts.find_each do |record|
+    known_names = fact_names.keys
```

Comment:
Did a little benchmarking, and looks like this caching is actually causing a performance loss:
```ruby
5000.times do
   a[SecureRandom.alphanumeric] = SecureRandom.base64
end  
keys = a.keys
Benchmark.ips do |x|
   x.report('key?'){a.key? 'aaaa'}
   x.report('keys.include'){keys.include? 'aaaa'}
   x.compare!
end 

Warming up --------------------------------------
                key?   374.975k i/100ms
        keys.include     1.897k i/100ms
Calculating -------------------------------------
                key?      8.248M (± 5.2%) i/s -     41.247M in   5.014675s
        keys.include     19.275k (± 1.8%) i/s -     96.747k in   5.021084s

Comparison:
                key?:  8247989.0 i/s
        keys.include:    19274.5 i/s - 427.92x  slower
```
This is because every `include?` check needs to iterate over the keys array rather than check the hash key map. It also adds an extra memory allocation for the keys array.

Reply:
Nice find, I did it for nothing actually, probably copy and paste.